spec.md

```diff
@@ -194,6 +194,31 @@ concurrently, including being both a producer and a consumer of events.
    For this, the framework will need a suitable metadata discriminator
    that frees it from having to understand the event details.
 
+## Attribute Categories
+
+The number of metadata attributes will be constrained to the minimal set needed
+to achieve the stated goals. To that end, attributes defined by this
+specification will fall into three categories:
+- required
+- optional
+- extensions
```

Comment:
nit: we seem to be dividing extensions into two categories as well:

* Registered extensions that vendors publish to encourage interoperability of a use case
* Private extensions

This concept was taken directly from the JWT spec where [private claims may be registered claims](https://tools.ietf.org/html/rfc7519#section-4.1).

Reply:
true - I guess originally I was thinking that we'd just talk about the stuff that might appear in our docs, but you're right, we should mention that people can do "other extensions" that this WG has no knowledge of.